### DISCUSSION TITLE
Add an Ability to Partially Download or Download all beatmaps at the Setup Wizard

### DISCUSSION THREAD

**Discussion by Shirou27** [ppy/osu#28405] [Ideas]
When a user factory resets their device due to random factors such as soft-bricking or boot-looping, their files will be lost, including those of osu!lazer. Downloading the beatmaps for osu!lazer is a hassle, especially if the user has many beatmaps. Therefore, the developers should implement the ability to download all the beatmaps based on the user's account history.

To do this, implement a button to setup wizard. After clicking that button, a login prompt will appear. If the game is interrupted or killed by the PC or Mobile OS while checking the email for verification code/link, the setup wizard should stay there when the user reopen the game.

**What if many users is doing the redownloading of their beatmaps?**

The downloader of the osu lazer should only be limited to one download and import at a time while the others is queued. Doing this not only removes the potential high load of the servers but also the potential crash of the game.

**What if the download usage is still high due to many users downloading at the same time?**

Limit the speed of the download and add a note or warning that the servers are in high load so the download will be limit and the download may take an hour or more to download all the beatmaps.

**What if the user can't wait it to download all the beatmaps and want to download it later?**

Add an ability to pause all the downloads.

**Comment by BlockHead007** [comment]
#27809, #27376 you're not alone
